autofill/autofill-impl/src/test/java/com/duckduckgo/autofill/ui/credential/management/CredentialGrouperTest.kt

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright (c) 2022 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.autofill.ui.credential.management
+
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import com.duckduckgo.autofill.AutofillDomainFormatterDomainNameOnly
+import com.duckduckgo.autofill.domain.app.LoginCredentials
+import com.duckduckgo.autofill.ui.credential.management.AutofillManagementRecyclerAdapter.ListItem
+import com.duckduckgo.autofill.ui.credential.management.AutofillManagementRecyclerAdapter.ListItem.Credential
+import com.duckduckgo.autofill.ui.credential.management.AutofillManagementRecyclerAdapter.ListItem.GroupHeading
+import org.junit.Assert.*
+import org.junit.Test
+import org.junit.runner.RunWith
+
+@RunWith(AndroidJUnit4::class)
+class CredentialGrouperTest {
+
+    private val characterValidator = LatinCharacterValidator()
+
+    private val initialExtractor = CredentialInitialExtractor(
+        domainFormatter = AutofillDomainFormatterDomainNameOnly(),
+        characterValidator = characterValidator
+    )
+
+    private val testee = CredentialGrouper(
+        initialExtractor = initialExtractor,
+        sorter = CredentialListSorterByTitleAndDomain(initialExtractor, characterValidator)
+    )
+
+    @Test
+    fun whenEmptyListInThenEmptyListOut() {
```

Comment:
Nit: Since this is more of an integration test, maybe we can add # cases as well.

Reply:
Missing case: A list with a credential with empty domain and empty title.